### PR TITLE
#20 Added cypress testing and multilingual

### PR DESCRIPTION
**Pull request by Rahul139-stack**
Added cypress testing  and please review it

### CODE REVIEW

**Comment by wildone** [comment]
lets remove [package-lock.json](https://github.com/innovolve-ai/template-electron-vuex-vuetify/pull/21/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519) as we have yarn.lock.

File: package.json [hunk 1]
```diff
@@ -1,35 +1,39 @@
-{
+ {
   "name": "template-electron-vuex-vuetify",
+  "version": "0.1.0",
```
Comment (wildone) [inline package.json:3]: this should just be what is develop.

File: package.json [hunk 2]
```diff
@@ -65,15 +70,13 @@
...
     "typescript": "~3.9.3",
     "vue-cli-plugin-electron-builder": "~2.0.0-rc.4",
+    "vue-cli-plugin-i18n": "^2.3.1",
     "vue-cli-plugin-vuetify": "~2.0.6",
     "vue-template-compiler": "^2.6.11",
     "vuetify-loader": "^1.3.0",
+    "webpack-dev-server": "^4.7.4",
     "xvfb-maybe": "^0.2.1"
   },
-  "electron": {
```
Comment (wildone) [inline package.json:73]: this sets size of window when app is open. do we need to remove?

File: package.json [hunk 3]
```diff
@@ -65,15 +70,13 @@
     "spectron": "^11.1.0",
     "typescript": "~3.9.3",
     "vue-cli-plugin-electron-builder": "~2.0.0-rc.4",
+    "vue-cli-plugin-i18n": "^2.3.1",
     "vue-cli-plugin-vuetify": "~2.0.6",
     "vue-template-compiler": "^2.6.11",
     "vuetify-loader": "^1.3.0",
+    "webpack-dev-server": "^4.7.4",
```
Comment (wildone) [inline package.json:77]: where is this used?

File: src/App.vue [hunk 4]
```diff
@@ -20,34 +20,39 @@
           width="100"
         />
       </div>
-
-      <v-spacer></v-spacer>
```
Comment (wildone) [inline src/App.vue:24]: any reason to remove this?

File: src/App.vue [hunk 5]
```diff
@@ -20,34 +20,39 @@
...
+        <span class="mr-2"
+          >{{ $t("components.language.appbar-end-text") }}
+        </span>
         <v-icon>mdi-open-in-new</v-icon>
       </v-btn>
     </v-app-bar>
 
     <v-main>
       <HelloWorld />
+      <!-- <Language></Language> -->
```
Comment (wildone) [inline src/App.vue:38]: do we need this?

File: src/App.vue [hunk 6]
```diff
@@ -20,34 +20,39 @@
...
 </template>
-
 <script lang="ts">
 import Vue from "vue";
 import HelloWorld from "./components/HelloWorld.vue";
+import Language from "./components/Language.vue";
 
 export default Vue.extend({
-  name: "App",
+  name: "App   ",
```
Comment (wildone) [inline src/App.vue:51]: is this right?

File: src/background.ts [hunk 7]
```diff
@@ -18,9 +18,16 @@ protocol.registerSchemesAsPrivileged([
 function createWindow() {
   // Create the browser window.
   win = new BrowserWindow({
+<<<<<<< Updated upstream
```
Comment (wildone) [inline src/background.ts:21]: this is a problem.

File: src/components/HelloWorld.vue [hunk 8]
```diff
@@ -6,33 +6,31 @@
           :src="require('../assets/logo.svg')"
           class="my-3"
           contain
-          height="200"
```
Comment (wildone) [inline src/components/HelloWorld.vue:9]: any reason for height change?

File: src/components/HelloWorld.vue [hunk 9]
```diff
@@ -6,33 +6,31 @@
...
-          <br />please join our online
-          <a href="https://community.vuetifyjs.com" target="_blank"
-            >Discord Community</a
-          >
-        </p>
+        <p
+          v-html="$t('components.helloworld.sub-header')"
+          class="subheading font-weight-regular"
+        ></p>
+        <!-- <h3>{{ $t('hello') }}</h3>
```
Comment (wildone) [inline src/components/HelloWorld.vue:21]: comments?

File: src/components/HelloWorld.vue [hunk 10]
```diff
@@ -83,61 +81,69 @@
...
-        text: "Select a layout",
-        href: "https://vuetifyjs.com/getting-started/pre-made-layouts"
-      },
-      {
-        text: "Frequently Asked Questions",
-        href: "https://vuetifyjs.com/getting-started/frequently-asked-questions"
-      }
-    ]
-  })
+  // created() {
```
Comment (wildone) [inline src/components/HelloWorld.vue:86]: comments?

File: src/components/Language.vue [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+<script>
+import { getModule } from "vuex-module-decorators";
+import Language from "../store/Modules/Language";
+const langModule = getModule(Language);
+
+export default {
+  methods: {
+    changeLanguage(val) {
+      console.log(val);
+      // this.$store.commit('setLanguage',val)
```
Comment (wildone) [inline src/components/Language.vue:29]: comment?

File: src/main.ts [hunk 12]
```diff
@@ -4,12 +4,16 @@ import "./registerServiceWorker";
 import router from "./router";
 import store from "./store";
 import vuetify from "./plugins/vuetify";
+import i18n from "./i18n";
 
 Vue.config.productionTip = false;
 
+console.log(store);
```
Comment (wildone) [inline src/main.ts:11]: ?

File: src/store/index.ts [hunk 13]
```diff
@@ -1,11 +1,47 @@
 import Vue from "vue";
 import Vuex from "vuex";
+// import Language from "@/store/modules/Language";
```
Comment (wildone) [inline src/store/index.ts:3]: How do we load this module?

File: src/store/index.ts [hunk 14]
```diff
@@ -1,11 +1,47 @@
...
+// const debug = process.env.NODE_ENV !== "production";
+
+const vuexLocal = new VuexPersistence({
+  storage: window.localStorage
 });
+
+export default new Vuex.Store({
+  plugins: [vuexLocal.plugin]
+  // strict: debug,
+}); // import Vue from "vue";
```
Comment (wildone) [inline src/store/index.ts:17]: do we need these comments?

File: tests/unit/example.spec.ts [hunk 15]
```diff
@@ -1,17 +1,17 @@
-import { shallowMount } from "@vue/test-utils";
-import HelloWorld from "@/components/HelloWorld.vue";
+// import { shallowMount } from "@vue/test-utils";
```
Comment (wildone) [inline tests/unit/example.spec.ts:1]: remove comments?

**Review by wildone — CHANGES_REQUESTED** [review]
looking good.

**Comment by wildone** [comment]
#20 add initial cypress configuration and tests for localization component.

File: package.json (hunk 3 above)
Comment (Rahul139-stack) [reply package.json:77]: this package is used for Cypress unit  testing configuration

File: src/App.vue (hunk 4 above)
Comment (Rahul139-stack) [reply src/App.vue:24]: Actually spacer tag is removed to fix the position of multilingual dropdown

File: src/store/index.ts (hunk 14 above)
Comment (Rahul139-stack) [reply src/store/index.ts:17]: In strict mode, whenever Vuex state is mutated outside of mutation handlers, an error will be thrown. This ensures that all state mutations can be explicitly tracked by debugging tools.

File: src/store/index.ts (hunk 13 above)
Comment (Rahul139-stack) [reply src/store/index.ts:3]: Using dynamic : true, Vuex allows us to register modules into store at runtime after store is constructed

File: src/App.vue (hunk 5 above)
Comment (Rahul139-stack) [reply src/App.vue:38]: Comment removed

File: src/App.vue (hunk 6 above)
Comment (Rahul139-stack) [reply src/App.vue:51]: Fixed

File: src/background.ts (hunk 7 above)
Comment (Rahul139-stack) [reply src/background.ts:21]: Fixed

File: src/components/HelloWorld.vue (hunk 8 above)
Comment (Rahul139-stack) [reply src/components/HelloWorld.vue:9]: Height of the vuetify image was too large, made it smaller

File: src/components/HelloWorld.vue (hunk 9 above)
Comment (Rahul139-stack) [reply src/components/HelloWorld.vue:21]: Comment removed

File: src/main.ts (hunk 12 above)
Comment (Rahul139-stack) [reply src/main.ts:11]: This was for debugging purpose. Removed

**Review by wildone — APPROVED** [review]

**Review by wildone — APPROVED** [review]

File: package.json [hunk 16]
```diff
@@ -1,35 +1,39 @@
-{
+ {
   "name": "template-electron-vuex-vuetify",
+  "version": "0.1.0",
+  "private": true,
   "description": "Description",
-  "productName": "Product Name",
```
Comment (wildone) [inline package.json:4]: we need a product Name for the apps as this appears in title bar and picked up by electron if we don't specify specific title in its config.

Comment (wildone) [reply package.json:4]: never mind its at the bottom.

**Pull request closed without merging**